src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/convert_pagedattn_inputs.cpp

```diff
@@ -0,0 +1,139 @@
+    auto result = pa_1 | pa_2;
+    auto key_cache_prec = config.keyCachePrecision;
+    auto value_cache_prec = config.valueCachePrecision;
+    auto key_cache_gs = config.keyCacheGroupSize;
+    auto value_cache_gs = config.valueCacheGroupSize;
+    auto infer_prec = config.inferencePrecision;
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto pa_op = m.get_match_root();
+        auto key_cache = ov::as_type_ptr<ov::op::v0::Parameter>(pa_op->get_input_node_shared_ptr(3));
+        auto value_cache = ov::as_type_ptr<ov::op::v0::Parameter>(pa_op->get_input_node_shared_ptr(4));
+        auto format_cache_precision = [](ov::element::Type cache_precision, ov::element::Type infer_precision) {
+            return cache_precision == ov::element::f16 && infer_precision == ov::element::bf16 ? infer_precision
+                                                                                               : cache_precision;
+        };
+        auto init_cache_shape = [&](const size_t head_nums,
+                                    const size_t head_size,
+                                    const ov::element::Type precision,
+                                    const size_t group_size,
+                                    const bool bychannel) {
+            ov::Dimension::value_type block_size = 32;
+            ov::Dimension::value_type _head_nums = head_nums;
+            ov::Dimension::value_type _head_size = head_size;
+            ov::Dimension::value_type _group_size = group_size;
+            _group_size = _group_size ? _group_size : _head_size;
+            if (!bychannel) {
+                if (_head_size % _group_size != 0) {
+                    OPENVINO_THROW("cache head_size ", head_size, "cannot be divided by group_size ", group_size);
+                }
+            }
+            size_t group_num = _head_size / _group_size;
+            if (precision == ov::element::u8) {
+                if (bychannel) {
+                    block_size += 2 * sizeof(float);
+                } else {
+                    _head_size += sizeof(float) * 2 * group_num;
+                }
+            } else if (precision == ov::element::u4) {
+                _head_size += sizeof(float) * 2 * group_num * 2;
+            }
+            return ov::PartialShape{-1, _head_nums, block_size, _head_size};
+        };
+        auto key_cache_precision = format_cache_precision(key_cache_prec, infer_prec);
+        auto value_cache_precision = format_cache_precision(value_cache_prec, infer_prec);
+        key_cache->set_element_type(key_cache_precision);
+        value_cache->set_element_type(value_cache_precision);
+        if (!pa_op->get_rt_info().count("num_k_heads") || !pa_op->get_rt_info().count("k_head_size") ||
+            !pa_op->get_rt_info().count("num_v_heads") || !pa_op->get_rt_info().count("num_v_heads")) {
+            OPENVINO_THROW("PagedAttn ",
```

Comment:
The `rt_info` may be valid if the client is genai, and if we want to be compatible with old vLLM, maybe no need to throw but log+return.

Reply:
applied in [27412e8](https://github.com/openvinotoolkit/openvino/pull/29018/commits/27412e8ecb5a83e5bf4863ad87bdcd1d29362002)